workflow/example.py

```diff
@@ -0,0 +1,180 @@
+import random
+import string
+
+import pydantic
+from workflow import slay
+
+########################################################################################
+
+
+class Parameters(pydantic.BaseModel):
+    length: int = 100
+    num_partitions: int = 4
+    num_replications: int = 3
+
+
+class WithParamsBase(pydantic.BaseModel):
+    params: Parameters
+
+
+class GenerateDataInput(WithParamsBase):
+    ...
+
+
+class GenerateDataOutput(WithParamsBase):
+    text: str
+
```

Comment:
These base classes maybe this whole situation a little too verbose. Should we just use normal data types, unless you need composite input?